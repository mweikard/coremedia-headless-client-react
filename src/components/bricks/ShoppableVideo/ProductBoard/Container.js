// @flow
import { Box } from '../../../basic/Box';
import withStyles from '../../../styles/withStyles';

const styles = (theme, props) => ({
  display: 'flex',
  flexWrap: 'nowrap',
  alignItems: 'center',
  height: '100%',
  transition: 'transform 0.45s ease-out',
  [theme.breakpoints.min('sm')]: {
    flexWrap: 'wrap',
    flexDirection: 'row',
    transform: 'unset',
    justifyContent: 'center',
    alignContent: props.productboardOverflow ? 'flex-start' : 'center',
  },
});

export default withStyles(styles, 'ShoppableVideoProductBoardContainer')(Box);
