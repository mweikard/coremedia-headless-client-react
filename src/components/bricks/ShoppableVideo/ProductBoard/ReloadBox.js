// @flow
import { Box } from '../../../basic/Box';
import withStyles from '../../../styles/withStyles';

const styles = (theme, props) => ({
  display: 'none',
  [theme.breakpoints.min('sm')]: {
    display: 'block',
    position: 'absolute',
    bottom: '3px',
    left: 0,
    zIndex: 9999,
  },
});

export default withStyles(styles, 'ShoppableVideoProductBoardReloadBox')(Box);
