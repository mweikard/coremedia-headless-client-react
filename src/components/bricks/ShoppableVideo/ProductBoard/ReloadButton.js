// @flow
import React from 'react';
import PropTypes from 'prop-types';

import { IconButton as BasicIconButton } from '../../../basic/Button';
import RepeatIcon from '../../../basic/Icon/RepeatIcon';
import withStyles from '../../../styles/withStyles';

const styles = (theme, props) => ({
  width: '100%',
  height: 'auto',
  [theme.breakpoints.min('sm')]: {
    width: '40px',
    height: '40px',
  },
});

const IconButton = withStyles(styles, 'ShoppableVideoProductBoardReloadButton')(BasicIconButton);

type Props = {
  id?: string,
  handleClick: (ev: SyntheticEvent<HTMLButtonElement>) => void,
};

const ReloadButton = ({ id, handleClick }: Props) => (
  <IconButton onClick={handleClick} color="#808080" backgroundColor="rgba(255, 255, 255, 0.5)">
    <RepeatIcon id={id} />
  </IconButton>
);

ReloadButton.propTypes = {
  id: PropTypes.string,
  handleClick: PropTypes.func.isRequired,
};

export default ReloadButton;
