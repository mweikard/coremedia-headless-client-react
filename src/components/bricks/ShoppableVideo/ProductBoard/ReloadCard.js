// @flow
import { Box } from '../../../basic/Box';
import withStyles from '../../../styles/withStyles';

const styles = (theme, props) => ({
  display: 'inline-flex',
  flex: '0 0 auto',
  position: 'relative',
  width: '144px',
  height: '173px',
  margin: '8px',
  padding: '0 36px',
  order: props.index,
  [theme.breakpoints.min('sm')]: {
    display: 'none',
  },
});

export default withStyles(styles, 'ShoppableVideoProductBoardReloadCard')(Box);
