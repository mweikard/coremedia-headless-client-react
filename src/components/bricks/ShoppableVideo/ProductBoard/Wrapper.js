// @flow
// import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';

import { getColors } from '../../../styles/utils';

const fadeIn = keyframes`
  from {
    opacity 0;
  }

  to {
    opacity 1;
  }
`;

const ProductBoardWrapper = styled.div`
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background-color: ${props => props.theme.palette.surface.color};
  overflow-x: scroll;
  overflow-y: hidden;
  direction: rtl;
  border-width: 3px 0;
  border-style: solid none;
  border-color: ${props => getColors(props.theme.palette, props.color).color};
  padding: 7px;
  animation: ${fadeIn} 0.6s linear;
  ${props => props.theme.breakpoints.min('sm')} {
    display: initial;
    overflow-x: hidden;
    overflow-y: scroll;
    direction: ltr;
  }
`;

ProductBoardWrapper.displayName = 'ProductBoardWrapper';

/*ProductBoardWrapper.propTypes = {
  theme: PropTypes.shape({
    breakpoints: PropTypes.shape({
      min: PropTypes.func.isRequired,
    }).isRequired,
  }),
};*/

ProductBoardWrapper.defaultProps = {
  color: 'primary',
  theme: {
    breakpoints: {
      min: () => `@media (min-width:${768 / 16}em)`,
    },
    palette: {
      surface: {
        color: '#fff',
      },
      primary: {
        color: 'rgb(0, 108, 174)',
        text: 'rgba(255,255,255,0.87)',
      },
    },
  },
};

export default ProductBoardWrapper;
