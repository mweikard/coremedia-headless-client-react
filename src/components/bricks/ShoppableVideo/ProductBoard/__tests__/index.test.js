// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import ProductBoard from '../';

const createMockMediaMatcher = matches => () => ({
  matches,
  addListener: () => {},
  removeListener: () => {},
});

describe('ProductBoard Component', () => {
  let originalMatchMedia;
  beforeEach(() => {
    originalMatchMedia = window.matchMedia;
  });

  afterEach(() => {
    window.matchMedia = originalMatchMedia;
  });

  it('should render correctly for breakpoint below sm', () => {
    window.matchMedia = createMockMediaMatcher(false);

    const thumbnails = [
      {
        id: 'ID1',
        teaserTitle: 'Title 1',
        teaserText: 'Text 1',
        pictureLink: 'Link 1',
        pictureTitle: 'Picture Title 1',
        pictureAlt: 'Picture Alt 1',
        price: 100,
        handleClick: jest.fn().mockName('handleClick'),
      },
      {
        id: 'ID2',
        teaserTitle: 'Title 2',
        teaserText: 'Text 2',
        pictureLink: 'Link 2',
        pictureTitle: 'Picture Title 2',
        pictureAlt: 'Picture Alt 2',
        price: 100,
        handleClick: jest.fn().mockName('handleClick'),
      },
      {
        id: 'ID3',
        teaserTitle: 'Title 3',
        teaserText: 'Text 3',
        pictureLink: 'Link 3',
        pictureTitle: 'Picture Title 3',
        pictureAlt: 'Picture Alt 3',
        price: 100,
        handleClick: jest.fn().mockName('handleClick'),
      },
    ];
    const handleReplay = jest.fn().mockName('handleReplay');
    const handleRef = jest.fn().mockName('handleRef');

    const wrapper = shallow(
      <ProductBoard
        thumbnails={thumbnails}
        handleReplay={handleReplay}
        handleRef={handleRef}
        productboardOverflow={false}
        theme={{
          breakpoints: {
            values: {
              sm: 768,
              md: 992,
              lg: 1280,
              xl: 1920,
            },
          },
        }}
      />
    );
    const boardWrapper = wrapper.find('ProductBoard').shallow();
    expect(boardWrapper).toMatchSnapshot();
    const mediaWrapper = boardWrapper.find('Media').shallow();
    expect(mediaWrapper).toMatchSnapshot();
    const transitionWrapper = mediaWrapper.find('Transition');
    expect(transitionWrapper).toHaveLength(3);
    transitionWrapper.forEach((node, index) => {
      expect(node.shallow()).toMatchSnapshot();
    });
  });

  it('should render correctly for breakpoint sm and above', () => {
    window.matchMedia = createMockMediaMatcher(true);

    const thumbnails = [
      {
        id: 'ID1',
        teaserTitle: 'Title 1',
        teaserText: 'Text 1',
        pictureLink: 'Link 1',
        pictureTitle: 'Picture Title 1',
        pictureAlt: 'Picture Alt 1',
        price: 100,
        handleClick: jest.fn().mockName('handleClick'),
      },
      {
        id: 'ID2',
        teaserTitle: 'Title 2',
        teaserText: 'Text 2',
        pictureLink: 'Link 2',
        pictureTitle: 'Picture Title 2',
        pictureAlt: 'Picture Alt 2',
        price: 100,
        handleClick: jest.fn().mockName('handleClick'),
      },
      {
        id: 'ID3',
        teaserTitle: 'Title 3',
        teaserText: 'Text 3',
        pictureLink: 'Link 3',
        pictureTitle: 'Picture Title 3',
        pictureAlt: 'Picture Alt 3',
        price: 100,
        handleClick: jest.fn().mockName('handleClick'),
      },
    ];
    const handleReplay = jest.fn().mockName('handleReplay');
    const handleRef = jest.fn().mockName('handleRef');

    const wrapper = shallow(
      <ProductBoard
        thumbnails={thumbnails}
        handleReplay={handleReplay}
        handleRef={handleRef}
        productboardOverflow={false}
        theme={{
          breakpoints: {
            values: {
              sm: 768,
              md: 992,
              lg: 1280,
              xl: 1920,
            },
          },
        }}
      />
    );
    const boardWrapper = wrapper.find('ProductBoard').shallow();
    expect(boardWrapper).toMatchSnapshot();
    const mediaWrapper = boardWrapper.find('Media').shallow();
    expect(mediaWrapper).toMatchSnapshot();
    const transitionWrapper = mediaWrapper.find('Transition');
    expect(transitionWrapper).toHaveLength(3);
    transitionWrapper.forEach((node, index) => {
      expect(node.shallow()).toMatchSnapshot();
    });
  });
});
