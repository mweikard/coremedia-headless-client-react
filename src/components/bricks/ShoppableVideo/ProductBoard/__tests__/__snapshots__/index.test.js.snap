// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductBoard Component should render correctly for breakpoint below sm 1`] = `
<React.Fragment>
  <ReloadBox
    theme={
      Object {
        "breakpoints": Object {
          "min": [Function],
        },
      }
    }
  >
    <ReloadButton
      handleClick={[MockFunction handleReplay]}
    />
  </ReloadBox>
  <ProductBoardWrapper
    color="primary"
    theme={
      Object {
        "breakpoints": Object {
          "min": [Function],
        },
        "palette": Object {
          "primary": Object {
            "color": "rgb(0, 108, 174)",
            "text": "rgba(255,255,255,0.87)",
          },
          "surface": Object {
            "color": "#fff",
          },
        },
      }
    }
  >
    <Media
      defaultMatches={true}
      query="(min-width: 768px)"
    />
  </ProductBoardWrapper>
</React.Fragment>
`;

exports[`ProductBoard Component should render correctly for breakpoint below sm 2`] = `
<TransitionGroup
  childFactory={[Function]}
  component={[Function]}
  innerRef={[MockFunction handleRef]}
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    key="ID3"
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  />
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    key="ID2"
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={500}
    unmountOnExit={false}
  />
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    key="ID1"
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={700}
    unmountOnExit={false}
  />
  <ProductBox
    index={0}
    theme={
      Object {
        "breakpoints": Object {
          "min": [Function],
        },
      }
    }
  >
    <ReloadButton
      handleClick={[MockFunction handleReplay]}
    />
  </ProductBox>
</TransitionGroup>
`;

exports[`ProductBoard Component should render correctly for breakpoint below sm 3`] = `
<ProductCard
  entered={false}
  handleClick={[MockFunction handleClick]}
  index={1}
  pictureAlt="Picture Alt 3"
  pictureLink="Link 3"
  pictureTitle="Picture Title 3"
  price={100}
  ratio="portrait_ratio5x6"
  title="Title 3"
/>
`;

exports[`ProductBoard Component should render correctly for breakpoint below sm 4`] = `
<ProductCard
  entered={false}
  handleClick={[MockFunction handleClick]}
  index={2}
  pictureAlt="Picture Alt 2"
  pictureLink="Link 2"
  pictureTitle="Picture Title 2"
  price={100}
  ratio="portrait_ratio5x6"
  title="Title 2"
/>
`;

exports[`ProductBoard Component should render correctly for breakpoint below sm 5`] = `
<ProductCard
  entered={false}
  handleClick={[MockFunction handleClick]}
  index={3}
  pictureAlt="Picture Alt 1"
  pictureLink="Link 1"
  pictureTitle="Picture Title 1"
  price={100}
  ratio="portrait_ratio5x6"
  title="Title 1"
/>
`;

exports[`ProductBoard Component should render correctly for breakpoint sm and above 1`] = `
<React.Fragment>
  <ReloadBox
    theme={
      Object {
        "breakpoints": Object {
          "min": [Function],
        },
      }
    }
  >
    <ReloadButton
      handleClick={[MockFunction handleReplay]}
    />
  </ReloadBox>
  <ProductBoardWrapper
    color="primary"
    theme={
      Object {
        "breakpoints": Object {
          "min": [Function],
        },
        "palette": Object {
          "primary": Object {
            "color": "rgb(0, 108, 174)",
            "text": "rgba(255,255,255,0.87)",
          },
          "surface": Object {
            "color": "#fff",
          },
        },
      }
    }
  >
    <Media
      defaultMatches={true}
      query="(min-width: 768px)"
    />
  </ProductBoardWrapper>
</React.Fragment>
`;

exports[`ProductBoard Component should render correctly for breakpoint sm and above 2`] = `
<TransitionGroup
  childFactory={[Function]}
  component={[Function]}
  innerRef={[MockFunction handleRef]}
  productboardOverflow={false}
>
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    key="ID1"
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  />
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    key="ID2"
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={500}
    unmountOnExit={false}
  />
  <Transition
    appear={true}
    enter={true}
    exit={true}
    in={false}
    key="ID3"
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={700}
    unmountOnExit={false}
  />
</TransitionGroup>
`;

exports[`ProductBoard Component should render correctly for breakpoint sm and above 3`] = `
<ProductCard
  entered={false}
  handleClick={[MockFunction handleClick]}
  index={0}
  pictureAlt="Picture Alt 1"
  pictureLink="Link 1"
  pictureTitle="Picture Title 1"
  price={100}
  ratio="portrait_ratio5x6"
  title="Title 1"
/>
`;

exports[`ProductBoard Component should render correctly for breakpoint sm and above 4`] = `
<ProductCard
  entered={false}
  handleClick={[MockFunction handleClick]}
  index={1}
  pictureAlt="Picture Alt 2"
  pictureLink="Link 2"
  pictureTitle="Picture Title 2"
  price={100}
  ratio="portrait_ratio5x6"
  title="Title 2"
/>
`;

exports[`ProductBoard Component should render correctly for breakpoint sm and above 5`] = `
<ProductCard
  entered={false}
  handleClick={[MockFunction handleClick]}
  index={2}
  pictureAlt="Picture Alt 3"
  pictureLink="Link 3"
  pictureTitle="Picture Title 3"
  price={100}
  ratio="portrait_ratio5x6"
  title="Title 3"
/>
`;
